File: typescript/tracing/anthropic-agents-sdk/hello_world_test.ts
/**
 * Hello World — Anthropic Agent SDK + Respan tracing.
 *
 * The simplest possible example: ask Claude a question, see the trace in Respan.
 *
 * Setup:
 *   npm install (or yarn install)
 *
 * Run:
 *   npx tsx hello_world_test.ts
 */

import "dotenv/config";
import { RespanAnthropicAgentsExporter } from "@respan/exporter-anthropic-agents";
import { queryForResult } from "./_sdk_runtime";

const API_KEY = process.env.RESPAN_API_KEY;
const BASE_URL = process.env.RESPAN_BASE_URL;

const exporter = new RespanAnthropicAgentsExporter({
  apiKey: API_KEY ?? undefined,
  endpoint: BASE_URL ? `${BASE_URL.replace(/\/+$/, "")}/v1/traces/ingest` : undefined,
});

async function main(): Promise<void> {
    console.log("Asking Claude a question...\n");

  if (!API_KEY) {
    throw new Error("Set RESPAN_API_KEY to run this example.");
  }

  const options = exporter.withOptions({
    permissionMode: "bypassPermissions",
    maxTurns: 1,
  });

  const { result, sessionId } = await queryForResult({
    prompt: "What is 2 + 2? Reply in one word.",
    options,
    onMessage: async (message, context) => {
      await exporter.trackMessage({ message, sessionId: context.sessionId });
      console.log(`  ${String(message.type ?? "unknown")}`);
    },
  });

  console.log(
    `  Result: subtype=${String(result.subtype)}, turns=${String(result.num_turns)}`,
  );

  console.log(`\nSession: ${sessionId}`);
  console.log("View trace at: https://platform.respan.ai/platform/traces");
}

main().catch(console.error);

File: python/tracing/anthropic-agents-sdk/basic/hello_world_test.py
"""
Hello World — Anthropic Agent SDK + Respan tracing.

The simplest possible example: ask Claude a question, see the trace in Respan.

Setup:
    pip install claude-agent-sdk respan-exporter-anthropic-agents python-dotenv

Run:
    python basic/hello_world_test.py

    # or with pytest:
    pytest basic/hello_world_test.py -v
"""

from dotenv import load_dotenv

load_dotenv(override=True)

import asyncio
import os

import pytest
from claude_agent_sdk import ClaudeAgentOptions

from respan_exporter_anthropic_agents import RespanAnthropicAgentsExporter
from _sdk_runtime import query_for_result

API_KEY = os.getenv("RESPAN_API_KEY") or os.getenv("RESPAN_API_KEY")
BASE_URL = os.getenv("RESPAN_BASE_URL") or os.getenv("RESPAN_BASE_URL")

# Create exporter — sends traces to Respan
exporter = RespanAnthropicAgentsExporter(
    api_key=API_KEY,
    base_url=BASE_URL,
)


@pytest.mark.asyncio
async def test_hello_world():
    """Ask Claude a simple question and export the trace."""

    options = ClaudeAgentOptions(
        permission_mode="bypassPermissions",
        max_turns=1,
    )

    result_message = await query_for_result(
        exporter=exporter,
        prompt="What is 2 + 2? Reply in one word.",
        options=options,
    )

    print(f"Result: {result_message.subtype}")
    print(f"Session: {exporter._last_session_id}")
    print("View trace at: https://platform.respan.ai/platform/traces")


if __name__ == "__main__":
    asyncio.run(test_hello_world())
